src/pages/OAuthRedirector.tsx

```diff
@@ -1,6 +1,16 @@
 import { OAUTH_PATH_STORAGE_KEY } from "constants/auth";
 import { Navigate } from "react-router-dom";
+import { StoredRouteState } from "types/auth";
 
 export default function OAuthRedirector() {
-  return <Navigate to={localStorage.getItem(OAUTH_PATH_STORAGE_KEY) ?? "/"} />;
+  const retrieved: StoredRouteState | null = (() => {
+    try {
+      const item = localStorage.getItem(OAUTH_PATH_STORAGE_KEY);
+      return item && JSON.parse(item);
+    } catch (_) {
+      return null;
```

Comment:
```suggestion
      return {};
```
nit: return empty object immediately, so there's no need for the null check below.